application-task-manager-default/src/main/java/com/xwiki/taskmanager/internal/DefaultTaskManager.java

```
@@ -0,0 +1,128 @@
+{
+    private static final LocalDocumentReference TASK_CLASS_REFERENCE = new LocalDocumentReference(Arrays.asList(
+        "TaskManager", "Code"), "TaskClass");
+
+    @Inject
+    private DocumentReferenceResolver<String> resolver;
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Inject
+    private QueryManager queryManager;
+
+    @Inject
+    private EntityReferenceSerializer<String> serializer;
+    @Inject
+    private Logger logger;
+
+    @Override
+    public Task getTaskByReference(String reference)
+    {
+        XWikiContext context = contextProvider.get();
+        DocumentReference docRef = resolver.resolve(reference);
+        try {
+            XWikiDocument doc = context.getWiki().getDocument(docRef, context);
+            Task task = new Task();
+            BaseObject obj = doc.getXObject(TASK_CLASS_REFERENCE);
+            if (obj == null) {
+                logger.warn("The page [{}] does not have a Task Object.", docRef);
+                return null;
+            }
+            task.setReference(docRef);
+            task.setName(obj.getStringValue(Task.NAME));
+            task.setNumber(obj.getIntValue(Task.NUMBER));
+            task.setOwner(resolver.resolve(obj.getLargeStringValue(Task.OWNER)));
+            task.setAssignee(resolver.resolve(obj.getLargeStringValue(Task.ASSIGNEE)));
+            task.setStatus(obj.getStringValue(Task.STATUS));
+            task.setReporter(resolver.resolve(obj.getLargeStringValue(Task.REPORTER)));
+            task.setRender(obj.getLargeStringValue(Task.RENDER));
+            task.setDuedate(obj.getDateValue(Task.DUE_DATE));
+            task.setCreateDate(obj.getDateValue(Task.CREATE_DATE));
+            task.setCompleteDate(obj.getDateValue(Task.COMPLETE_DATE));
+            return task;
+        } catch (XWikiException e) {
+            logger.error("Failed to retrieve the task from the page [{}]", reference);
+            return null;
```

It's usually better to let the user of the API decide if it fails silently or not. I'd throw a ``TaskException`` here because failing to read the task object is not expected. Also, for the user of the API it's not possible currently to know if the return ``null`` value is because the task object is missing or because there's a problem with the task object (that prevents us to read it).